### PR TITLE
Update NNPI Backend to v0.6.1.0

### PR DESCRIPTION
**Pull request by omromano**
Summary: Updating to NNPI Backend version v0.6.1.0

### CODE REVIEW

**Comment by omromano** [comment]
Hi @jfix71, @arunm-git 
Please take a look.
CC: @jsubag

File: lib/Backends/NNPI/NNPI.cpp [hunk 1]
```diff
@@ -1298,6 +1308,14 @@ Error NNPIBackend::bindContexts(
       }
     }
     if (ctx && devMgr) {
+      // Update the tensors bound to placeholders.
+      auto phBindings = ctx->getPlaceholderBindings();
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPI.cpp:1312]: nit: `auto *phBindings`

File: lib/Backends/NNPI/NNPI.cpp [hunk 2]
```diff
@@ -1313,3 +1331,260 @@ Error NNPIBackend::bindContexts(
 
   return Error::success();
 }
+
+// Partial update of the NNPITensorDesc. Some members are ignored as they're
+// not used for estimation.
+static bool UpdateDescForEstimate(NNPITensorDesc &desc,
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPI.cpp:1337]: nit: `updateDescForEstimate `

File: lib/Backends/NNPI/NNPI.cpp [hunk 3]
```diff
@@ -1313,3 +1331,260 @@ Error NNPIBackend::bindContexts(
...
+  memset(&desc.quantParams.params, 0,
+         sizeof(desc.quantParams.params)); // Actual values are not needed here.
+
+  desc.attributes.value = 0; // No attributes needed here.
+
+  return true;
+}
+
+// Prepare the list of NNPITensorDesc for the estimate call.
+static bool UpdateDescListForEstimate(std::vector<NNPITensorDesc> &descs,
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPI.cpp:1408]: nit: `updateDescListForEstimate`

File: lib/Backends/NNPI/NNPI.cpp [hunk 4]
```diff
@@ -1313,3 +1331,260 @@ Error NNPIBackend::bindContexts(
...
+  }
+  memset(&desc.quantParams.params, 0,
+         sizeof(desc.quantParams.params)); // Actual values are not needed here.
+
+  desc.attributes.value = 0; // No attributes needed here.
+
+  return true;
+}
+
+// Prepare the list of NNPITensorDesc for the estimate call.
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPI.cpp:1407]: nit: `///`

File: lib/Backends/NNPI/NNPI.cpp [hunk 5]
```diff
@@ -1313,3 +1331,260 @@ Error NNPIBackend::bindContexts(
 
   return Error::success();
 }
+
+// Partial update of the NNPITensorDesc. Some members are ignored as they're
+// not used for estimation.
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPI.cpp:1336]: nit: `///`

File: lib/Backends/NNPI/NNPI.cpp [hunk 6]
```diff
@@ -1313,3 +1331,260 @@ Error NNPIBackend::bindContexts(
...
+    // Op isn't supported.
+    return -1.0;
+  }
+  NNPI_LENGTH_TYPE lengthType = NNPI_LENGTH_VARIABLE;
+  LOG_AND_RETURN_IF(ERROR,
+                    NNPIImporter::convertLengthsModeToLengthType(
+                        lengthsMode, lengthType) != NNPI_NO_ERROR,
+                    "Failed to convert LengthsMode", -1.0);
+
+  enum DescIndex {
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPI.cpp:1436]: Unused?

File: lib/Backends/NNPI/NNPI.cpp (hunk 3 above)
Comment (jfix71) [inline lib/Backends/NNPI/NNPI.cpp:1408]: nit: `const std::vector`

File: lib/Backends/NNPI/NNPI.cpp (hunk 2 above)
Comment (jfix71) [reply lib/Backends/NNPI/NNPI.cpp:1337]: nit: `const NNPITensorDesc &`

File: lib/Backends/NNPI/NNPI.cpp (hunk 6 above)
Comment (jsubag) [reply lib/Backends/NNPI/NNPI.cpp:1436]: Implicitly used in the descs vector.
i.e. the first one is for input, the second is for output etc.

File: lib/Backends/NNPI/Importer.cpp [hunk 7]
```diff
@@ -56,12 +56,8 @@ static std::string nodeValueName(const glow::NodeValue &nv) {
          std::to_string(nv.getResNo());
 }
 
-static inline NNPIErrorCode
-convertLengthsModeToLengthType(glow::LengthsMode mode,
-                               NNPI_LENGTH_TYPE &lengthType) {
-  if (!GlowNNPISpecializeAllOneSLS) {
-    mode = LengthsMode::Variable;
-  }
```
[the remark this answers is not in the document]
Comment (omromano) [reply lib/Backends/NNPI/Importer.cpp:64]: Sorry - I missed that in the merge (in the GitHub tool... )

File: lib/Backends/NNPI/NNPI.cpp (hunk 2 above)
Comment (omromano) [reply lib/Backends/NNPI/NNPI.cpp:1337]: Changed the method name but the cant change to const as desc is updated in this method.

File: lib/Backends/NNPI/NNPI.cpp (hunk 6 above)
Comment (jfix71) [reply lib/Backends/NNPI/NNPI.cpp:1436]: Maybe makes sense to move it down to just before `nnpiEstimateSparseLengthsWeightedSumOp()`. Or even just remove it and use `/* Input */ 0` in the function call. No biggy though.

Comment (omromano) [reply lib/Backends/NNPI/NNPI.cpp:1436]: It is used see line #1581.

**Pull request closed without merging**